pkg/operator/ceph/cluster/mon/drain.go

```diff
@@ -52,22 +54,45 @@ func (c *Cluster) reconcileMonPDB() error {
 }
 
 func (c *Cluster) createOrUpdateMonPDB(maxUnavailable int32) (controllerutil.OperationResult, error) {
-	pdb := &policyv1beta1.PodDisruptionBudget{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      monPDBName,
-			Namespace: c.Namespace,
-		}}
+	k8sVersion, err := k8sutil.GetK8SVersion(c.context.Clientset)
+	if err != nil {
+		return controllerutil.OperationResultNone, errors.Wrap(err, "failed to fetch get k8s version")
+	}
+	// minimum k8s version required for v1 PodDisruptionBudget is 'v1.21.0'. Apply v1 if k8s version is at least 'v1.21.0', else apply v1beta1 PodDisruptionBudgetJob.
+	minVersionForPDBV1 := "1.21.0"
+	usePDBV1 := k8sVersion.AtLeast(version.MustParseSemantic(minVersionForPDBV1))
+	objectMeta := metav1.ObjectMeta{
+		Name:      monPDBName,
+		Namespace: c.Namespace,
+	}
+	selector := &metav1.LabelSelector{
+		MatchLabels: map[string]string{k8sutil.AppAttr: AppName},
+	}
+	if usePDBV1 {
+		pdb := &policyv1.PodDisruptionBudget{
+			ObjectMeta: objectMeta}
 
-	mutateFunc := func() error {
-		pdb.Spec = policyv1beta1.PodDisruptionBudgetSpec{
-			Selector: &metav1.LabelSelector{
-				MatchLabels: map[string]string{k8sutil.AppAttr: AppName},
-			},
-			MaxUnavailable: &intstr.IntOrString{IntVal: maxUnavailable},
+		mutateFunc := func() error {
+			pdb.Spec = policyv1.PodDisruptionBudgetSpec{
+				Selector:       selector,
+				MaxUnavailable: &intstr.IntOrString{IntVal: maxUnavailable},
+			}
+			return nil
 		}
-		return nil
+		return controllerutil.CreateOrUpdate(context.TODO(), c.context.Client, pdb, mutateFunc)
+	} else {
```

Comment:
nit: since there is a return from the previous `if` block, no need for the `else` block since it's implicit.
```suggestion
	{
```